### PR TITLE
msu_package: Fix cumulative updates installation and provide a 3600s default timeout

### PR DESCRIPTION
**Pull request by dheerajd-msys**
Signed-off-by: dheerajd-msys <dheeraj.dubey@msystechnologies.com>

## Description

-- Updated to match with version as well along with package name.
-- Identifies correct match for cumulative updates.
-- Added `timeout` property for setting new default 3600 because some packages which are larger in sizes gets time nearly 45mins to 60mins to complete.
## Related Issue

Fixes: https://github.com/chef/chef/issues/9383
Fixes: #9245 
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

**Comment by Vasu1105** [comment]
According to me we should cover the test for the version for which we made the changes. I haven't saw that part has been covered in specs.

File: lib/chef/provider/package/cab.rb [hunk 1]
```diff
@@ -74,7 +74,7 @@ def remove_package(name, version)
 
         def dism_command(command)
           with_os_architecture(nil) do
-            result = shell_out("dism.exe /Online /English #{command} /NoRestart", { timeout: new_resource.timeout })
+            result = shell_out("dism.exe /Online /English #{command} /NoRestart", { timeout: new_resource.timeout || 3600 })
```
Comment (dheerajd-msys) [inline lib/chef/provider/package/cab.rb:77]: I am only concerned about timeout to be ORed 3600. During testing on 2016 I found the package to get installed in nearly 45mins. So I've added here.

Comment (tas50) [reply lib/chef/provider/package/cab.rb:77]: If we want to change the default we should define a timeout property in the msu_package resource and set a new default. Doing it this way hides it from our documentation generation so no user will know there's a new default.

Comment (dheerajd-msys) [reply lib/chef/provider/package/cab.rb:77]: Thanks @tas50 for suggestion. Will be adding `timeout` property to override default.

**Comment by dheerajd-msys** [comment]
The msu package with which we got the issue reproduced is larger in size and we wont be able to cover in the test case including that package as source.

**Pull request merged**